### PR TITLE
Add test for randomize_genome function of individual classes

### LINKED ISSUE #186: Add test for `randomize_genome` of individual classes
This function is currently untested, we should add a test for it.

### PR DESCRIPTION
**Pull request by mschmidt87**
Adds a simple test for the `randomize_genome` function of the individual classes. All it tests is that the `randomize_genome` function does indeed change the dna.

Closes #186

### CODE REVIEW

File: test/test_individual.py [hunk 1]
```diff
@@ -294,3 +295,17 @@ def test_update_parameters_from_torch_class_does_not_reset_fitness_for_unused_pa
...
+
+
+@pytest.mark.parametrize("individual_type", ["SingleGenome", "MultiGenome"])
+def test_individual_randomize_genome(individual_type, rng_seed):
+    rng = np.random.RandomState(rng_seed)
+    primitives = (cgp.Add, cgp.Mul)
+    genome = cgp.Genome(1, 1, 2, 1, 1, primitives)
+    genome.randomize(rng)
+
+    dna_old = copy.copy(genome.dna)
```
Comment (jakobj) [inline test/test_individual.py:307]: i think `list(genome.dna)` would also work and avoid the import.

**Comment by jakobj** [comment]
oh and i'm confused how this "closes" itself? ;)

**Comment by mschmidt87** [comment]
> oh and i'm confused: how does this "close" itself? ;)

??

**Comment by mschmidt87** [comment]
>  i am wondering however, whether this would not better fit into `test_genome` the individuals are just forwarding everything to the genome aren't they? furthermore, from a quick grep it seems like `randomize_genome` is never used?! is this possible? and if so, shouldn't we rather remove it to keep confusion levels low?

My main motivation here was to not leave the code untested, but yeah, this function is pretty redundant and if it not used, we can also remove it altogether, I agree. If that is our decision, we can close this PR and I'll open another one tomorrow, removing this functionality from the classes.

**Comment by mschmidt87** [comment]
On 2nd thought, I think that we should instead of removing the function, rather use it in the `Population` in the `generate_random_individual`. Currently, we are doing a rather convoluted type check which is necessary because we call the `genome.randomize()` method. If instead, we call `individual.randomize_genome()`, this should become much easier.

**Comment by mschmidt87** [comment]
I implemented the change in `Population`

**Comment by mschmidt87** [comment]
Please take another look @jakobj

File: test/test_individual.py (hunk 1 above)
Comment (jakobj) [reply test/test_individual.py:307]: (bump)

Comment (mschmidt87) [reply test/test_individual.py]: Ah thanks, I had forgotten about that. Fixed now.

**Comment by mschmidt87** [comment]
Yep, thanks, I had forgotten about that, fixed it now and already squashed commits to 2 commits. @jakobj

**Pull request merged**